### PR TITLE
feat: customLabels in ComponentSpec

### PR DESCRIPTION
**Pull request by ShouEnHsiao**
This change allow user to inject
customized label to deployment.

### CODE REVIEW

**Review by thcdrt — APPROVED** [review]

File: apis/meta/v1alpha1/component.go [hunk 1]
```diff
@@ -119,6 +123,20 @@ func (c *ComponentSpec) ApplyToDeployment(deploy *appsv1.Deployment) {
 	deploy.Spec.Template.Spec.ImagePullSecrets = c.ImagePullSecrets
 	deploy.Spec.Template.Spec.NodeSelector = c.NodeSelector
 	deploy.Spec.Template.Spec.Tolerations = c.Tolerations
+	deploy.Spec.Template.Labels = mergeLabels(deploy.Spec.Template.Labels, c.TemplateLabels)
+}
+
+// mergeLabels merges all labels together and returns a new label.
+func mergeLabels(allLabels ...map[string]string) map[string]string {
```
[the remark this answers is not in the document]
Comment (ShouEnHsiao) [reply apis/meta/v1alpha1/component.go:130]: Do you have any recommended location? Or I can put all back in ApplyToDeployment so it does not looks random? 
Maybe something like this 
```
func (c *ComponentSpec) ApplyToDeployment(deploy *appsv1.Deployment) {
	deploy.Spec.Replicas = c.Replicas
	deploy.Spec.Template.Spec.ServiceAccountName = c.ServiceAccountName

	for i := range deploy.Spec.Template.Spec.Containers {
		if c.ImagePullPolicy != nil {
			deploy.Spec.Template.Spec.Containers[i].ImagePullPolicy = *c.ImagePullPolicy
		} else {
			deploy.Spec.Template.Spec.Containers[i].ImagePullPolicy = config.DefaultImagePullPolicy
		}

		deploy.Spec.Template.Spec.Containers[i].Resources = c.Resources
	}

	for k, v := range c.TemplateLabels {
		deploy.Spec.Template.Labels[k] = v
	}

	deploy.Spec.Template.Spec.ImagePullSecrets = c.ImagePullSecrets
	deploy.Spec.Template.Spec.NodeSelector = c.NodeSelector
	deploy.Spec.Template.Spec.Tolerations = c.Tolerations
}
```

[the remark this answers is not in the document]
Comment (MarcelMue) [reply apis/meta/v1alpha1/component.go:130]: I agree - including it in the `ApplyToDeployment` function makes sense to me.

[the remark this answers is not in the document]
Comment (ShouEnHsiao) [reply apis/meta/v1alpha1/component.go]: Thanks. Updated this commit as discussed.

**Comment by MarcelMue** [comment]
Thanks for your PR :)

**Pull request merged**